contracts/test/SafeSecretHarbour.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { loadFixture } from "@nomicfoundation/hardhat-toolbox/network-helpers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { randomX25519KeyPair } from "./utils/encryption";
+
+describe("SafeInternationalHarbour", () => {
+	async function deployFixture() {
+		const [deployer, signer, alice] = await ethers.getSigners();
+		const Factory = await ethers.getContractFactory("SafeSecretHarbour");
+		const harbour = await Factory.deploy();
+
+		const { decryptionKey, encryptionKey } = await randomX25519KeyPair();
+
+		return { deployer, signer, alice, harbour, decryptionKey, encryptionKey };
+	}
+
+	it("should register a public encryption key", async () => {
```

Comment:
IMO, this test primarily checks if the event is raised, rather than verifying the storage of the encryption key.

Reply:
Agree, probably should also do the retrieve or use getStorageAt to check the storage if we want to really check if it was stored ... or just rename the test to "should emit event when registering public encryption key" 😅

Reply:
Split this into two tests: one that reads the storage, the other that checks for the event.